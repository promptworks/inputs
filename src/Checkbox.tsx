import * as React from "react";
import { useField } from "./useField";
import { CheckboxProps, Element, GetProps } from "./types";

export function createCheckbox<E>(getProps: GetProps<CheckboxProps, E>) {
  return function Checkbox(props: CheckboxProps & E): Element {
    const field = useField(getProps(props));

    const onChange = React.useCallback(
      (event: React.ChangeEvent<HTMLInputElement>) => {
        field.onChange(event.target.checked);
      },
      [field.onChange]
    );

    return (
      <div {...field.getContainerProps()}>
        <input
          {...field.getInputProps()}
          type="checkbox"
          checked={field.value}
          onChange={onChange}
        />

        {field.label && (
          <label {...field.getLabelProps()}>
            {field.label}
            {field.help && <span {...field.getHelpProps()}>{field.help}</span>}
          </label>
        )}

        {field.error && <span {...field.getErrorProps()}>{field.error}</span>}
      </div>
    );
  };
}

export const Checkbox = createCheckbox(props => props);
