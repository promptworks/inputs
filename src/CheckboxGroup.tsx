import * as React from "react";
import { FieldProps, useField } from "./useField";
import { useConfig, Config } from "./useConfig";
import { FieldSetProps, useFieldSet } from "./useFieldSet";
import {
  HTMLProps,
  contains,
  createGroupContext,
  getDOMValue,
  isUndefined,
  toggle
} from "./utilities";

export interface CheckboxGroupProps<V> extends FieldSetProps {
  name: string;
  value?: V[];
  children?: React.ReactNode;
  onChangeValue?: (value: V[]) => void;
}

export interface CheckboxGroupItemProps
  extends FieldProps,
    HTMLProps<HTMLInputElement> {
  value: any;
}

export function createCheckboxGroup<T>(
  config: Config<CheckboxGroupProps<any>, T>
) {
  const { Provider, useGroup } = createGroupContext("CheckboxGroup");

  function CheckboxGroup(props: CheckboxGroupProps<any> & T) {
    const { children, ...otherProps } = useConfig(config, props);
    const field = useFieldSet(otherProps);

    return (
      <fieldset {...field.getFieldSetProps()}>
        {field.legend && (
          <legend {...field.getLegendProps()}>{field.legend}</legend>
        )}

        {field.help && <span {...field.getHelpProps()}>{field.help}</span>}
        <Provider value={field.getFieldProps()}>{children}</Provider>
        {field.error && <span {...field.getErrorProps()}>{field.error}</span>}
      </fieldset>
    );
  }

  function Item(props: CheckboxGroupItemProps) {
    const { value: selected, onChangeValue, ...groupProps } = useGroup();
    const { value, onChange, ...otherProps } = { ...groupProps, ...props };
    const field = useField(otherProps);
    const checked = isUndefined(selected)
      ? undefined
      : contains(selected, value);

    const handleChange = React.useCallback(
      (event: React.ChangeEvent<HTMLInputElement>) => {
        onChange && onChange(event);
        onChangeValue && onChangeValue(toggle(selected, value));
      },
      [value, selected, onChange, onChangeValue]
    );

    return (
      <div {...field.getFieldProps()}>
        <input
          type="checkbox"
          checked={checked}
          value={getDOMValue(value)}
          onChange={handleChange}
          {...field.getInputProps()}
        />

        {field.label && (
          <label {...field.getLabelProps()}>
            {field.label}
            {field.help && <span {...field.getHelpProps()}>{field.help}</span>}
          </label>
        )}

        {field.error && <span {...field.getErrorProps()}>{field.error}</span>}
      </div>
    );
  }

  return Object.assign(CheckboxGroup, { Item });
}

export const CheckboxGroup = createCheckboxGroup({});
