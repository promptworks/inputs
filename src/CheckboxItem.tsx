import * as React from "react";
import { FieldProps } from "./types";
import { contains, getDOMValue, remove, useNestedId } from "./utilities";
import { useFieldProps } from "./useFieldProps";

export interface CheckboxItemProps<Value>
  extends FieldProps<Value[], HTMLInputElement> {
  value: Value;
}

export function CheckboxItem<Value>(props: CheckboxItemProps<Value>) {
  const nested = { ...props, field: useNestedId(props.field) };
  const {
    field,
    label,
    help,
    error,
    append,
    prepend,
    value: item,
    getFieldProps,
    getLabelProps,
    getErrorProps,
    getHelpProps,
    ...inputProps
  } = useFieldProps(nested, "check", "checkbox");

  const { value, setValue } = field;
  const onChange = React.useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      const { checked } = event.target;
      setValue(value => (checked ? [...value, item] : remove(value, item)));
    },
    [item, setValue]
  );

  return (
    <div {...getFieldProps()}>
      {prepend}
      <input
        {...inputProps}
        type="checkbox"
        onChange={onChange}
        value={getDOMValue(item)}
        checked={contains(value, item)}
      />
      {append}
      <label {...getLabelProps()}>
        {label}
        {help && <span {...getHelpProps()}>{help}</span>}
      </label>
      {error && <span {...getErrorProps()}>{error}</span>}
    </div>
  );
}
