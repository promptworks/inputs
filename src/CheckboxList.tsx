import * as React from "react";
import { Field } from "./Field";
import { CheckboxListProps, CheckboxListOptionProps } from "./types";

const getCheckboxProps = (option: CheckboxListOptionProps | string) => {
  if (typeof option === "string") {
    return { value: option, key: option, label: option };
  }

  const { value, label = value, key = value, ...props } = option;
  return { value, label, key, ...props };
};

const remove = (items: any[], item: any) => {
  const result = items.concat([]);
  result.splice(items.indexOf(item), 1);
  return result;
};

export const CheckboxList: React.FC<CheckboxListProps> = ({
  value,
  onChange,
  options = [],
  ...props
}) => {
  const values = value || [];
  const handleChange = React.useCallback(
    event => {
      const { checked, value } = event.target;

      if (checked) {
        onChange([...values, value]);
      } else {
        onChange(remove(values, value));
      }
    },
    [onChange]
  );

  return (
    <>
      {options.map(option => {
        const checkbox = getCheckboxProps(option);

        return (
          <Field
            inline
            key={checkbox.key}
            label={checkbox.label}
            disabled={checkbox.disabled}
            render={inputProps => (
              <input
                {...inputProps}
                type="checkbox"
                checked={values.includes(checkbox.value)}
                value={checkbox.value}
                onChange={handleChange}
              />
            )}
            {...props}
          />
        );
      })}
    </>
  );
};
