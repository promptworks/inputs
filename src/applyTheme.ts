import { ClassName, Theme } from "./types";

function compileClassName<Props>(
  classNames: string | ClassName<Props> | undefined,
  props: Props,
  className: string = ""
): string | undefined {
  if (typeof classNames === "string") {
    className += ` ${classNames}`;
  } else if (classNames) {
    for (const key in classNames) {
      let test = classNames[key];
      if (typeof test === "function") test = test(props);
      if (test) className += ` ${key}`;
    }
  }

  return className.trim() || undefined;
}

export function applyTheme<Props extends { [key: string]: any }, ThemeProps>(
  props: Props & ThemeProps,
  theme: Theme<ThemeProps, Props>
): Props {
  const newProps = { ...props };
  const { props: themeProps = [], ...classNames } = theme;

  for (const prop in classNames) {
    const spec = (classNames as any)[prop];
    const className = compileClassName(spec, props, props[prop]);

    if (className) {
      (newProps as any)[prop] = className;
    }
  }

  for (const prop of themeProps) {
    delete newProps[prop];
  }

  return newProps;
}
