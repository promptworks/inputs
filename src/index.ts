export * from "./Checkbox";
export * from "./CheckboxList";
export * from "./Field";
export * from "./DateInput";
export * from "./DateTimeInput";
export * from "./defaultTheme";
export * from "./FileInput";
export * from "./FileListInput";
export * from "./FloatInput";
export * from "./IntegerInput";
export * from "./RadioGroup";
export * from "./Select";
export * from "./TextArea";
export * from "./TextInput";
export * from "./ToggleButton";
export * from "./types";
