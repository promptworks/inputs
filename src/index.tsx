export * from "./Checkbox";
export * from "./DateInput";
export * from "./DateTimeInput";
export * from "./FileInput";
export * from "./FloatInput";
export * from "./Input";
export * from "./IntegerInput";
export * from "./Select";
