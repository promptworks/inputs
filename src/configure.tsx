import * as React from "react";
import { Input } from "./Input";
import { Select } from "./Select";
import { Checkbox } from "./Checkbox";
import { CheckboxItem } from "./CheckboxItem";
import { TextArea } from "./TextArea";
import { Radio } from "./Radio";
import { FieldSet } from "./FieldSet";
import {
  FC,
  Config,
  Components,
  InputProps,
  SelectProps,
  CheckboxProps,
  CheckboxItemProps,
  TextAreaProps,
  RadioProps,
  FieldSetProps
} from "./types";

export function withProps<Props, ExtraProps>(
  Inner: FC<Props>,
  displayName: string,
  getProps: (props: Props & ExtraProps) => Props
): FC<Props & ExtraProps> {
  const Outer: FC<Props & ExtraProps> = props => {
    return <Inner {...getProps(props)} />;
  };

  if (Inner.displayName) {
    Outer.displayName = `${displayName}${Inner.displayName}`;
  }

  return Outer;
}

export function configure<ExtraProps>(
  config: Config<ExtraProps>
): Components<ExtraProps> {
  return {
    Input: withProps<InputProps, ExtraProps>(
      Input,
      config.displayName,
      config.getInputProps
    ),
    Select: withProps<SelectProps, ExtraProps>(
      Select,
      config.displayName,
      config.getSelectProps
    ),
    Checkbox: withProps<CheckboxProps, ExtraProps>(
      Checkbox,
      config.displayName,
      config.getCheckboxProps
    ),
    CheckboxItem: withProps<CheckboxItemProps<any>, ExtraProps>(
      CheckboxItem,
      config.displayName,
      config.getCheckboxItemProps
    ),
    TextArea: withProps<TextAreaProps, ExtraProps>(
      TextArea,
      config.displayName,
      config.getTextAreaProps
    ),
    Radio: withProps<RadioProps<any>, ExtraProps>(
      Radio,
      config.displayName,
      config.getRadioProps
    ),
    FieldSet: withProps<FieldSetProps, ExtraProps>(
      FieldSet,
      config.displayName,
      config.getFieldSetProps
    )
  };
}
