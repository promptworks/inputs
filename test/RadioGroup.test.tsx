import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import { itBehavesLikeARadioGroup } from "./sharedExamples";
import { RadioGroup, RadioGroupProps } from "../src";

const setup = (props: Partial<RadioGroupProps> = {}) =>
  render(
    <RadioGroup
      inline
      title="Jawn"
      name="jawn"
      options={[
        { label: "Foo", value: "foo" },
        { label: "Bar", value: "bar" },
        { label: "Disabled", value: "buzz", disabled: true }
      ]}
      onChange={jest.fn()}
      {...props}
    />
  );

describe("<RadioGroup />", () => {
  itBehavesLikeARadioGroup(setup);

  test("emits the value on change", () => {
    const onChange = jest.fn();
    const { getByLabelText } = setup({ onChange });

    fireEvent.change(getByLabelText("Foo"), {
      target: { value: "foo" }
    });

    expect(onChange).toHaveBeenCalledWith("foo");
  });

  test("emits `null` when the value is blank", () => {
    const onChange = jest.fn();
    const { getByLabelText } = setup({ onChange });

    fireEvent.change(getByLabelText("Foo"), {
      target: { value: "" }
    });

    expect(onChange).toHaveBeenCalledWith(null);
  });
});
