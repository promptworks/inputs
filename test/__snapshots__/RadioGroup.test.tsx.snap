// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_1"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline"
        for="field_1"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_2"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline"
        for="field_2"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--disabled"
        disabled=""
        id="field_3"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--disabled"
        for="field_3"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with a custom ID 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="foo"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline"
        for="foo"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="foo"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline"
        for="foo"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--disabled"
        disabled=""
        id="foo"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--disabled"
        for="foo"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with an error 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--erroneous"
    >
      <input
        aria-labelledby="field_4_label field_4_error"
        class="field__input field__input--inline field__input--erroneous"
        id="field_4"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--erroneous"
        for="field_4"
        id="field_4_label"
      >
        Foo
      </label>
      <span
        class="message message--problem"
        id="field_4_error"
        role="alert"
      >
        Oh no!
      </span>
    </div>
    <div
      class="field field--inline field--erroneous"
    >
      <input
        aria-labelledby="field_5_label field_5_error"
        class="field__input field__input--inline field__input--erroneous"
        id="field_5"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--erroneous"
        for="field_5"
        id="field_5_label"
      >
        Bar
      </label>
      <span
        class="message message--problem"
        id="field_5_error"
        role="alert"
      >
        Oh no!
      </span>
    </div>
    <div
      class="field field--inline field--erroneous field--disabled"
    >
      <input
        aria-labelledby="field_6_label field_6_error"
        class="field__input field__input--inline field__input--erroneous field__input--disabled"
        disabled=""
        id="field_6"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--erroneous field__label--disabled"
        for="field_6"
        id="field_6_label"
      >
        Disabled
      </label>
      <span
        class="message message--problem"
        id="field_6_error"
        role="alert"
      >
        Oh no!
      </span>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with help 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_7"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline"
        for="field_7"
      >
        Foo
        <span
          class="field__help"
        >
          Good luck
        </span>
      </label>
    </div>
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_8"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline"
        for="field_8"
      >
        Bar
        <span
          class="field__help"
        >
          Good luck
        </span>
      </label>
    </div>
    <div
      class="field field--inline field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--disabled"
        disabled=""
        id="field_9"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--disabled"
        for="field_9"
      >
        Disabled
        <span
          class="field__help"
        >
          Good luck
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with inline 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_22"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline"
        for="field_22"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_23"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline"
        for="field_23"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--disabled"
        disabled=""
        id="field_24"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--disabled"
        for="field_24"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with large 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--large"
    >
      <input
        class="field__input field__input--inline field__input--large"
        id="field_16"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--large"
        for="field_16"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--large"
    >
      <input
        class="field__input field__input--inline field__input--large"
        id="field_17"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--large"
        for="field_17"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--large field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--large field__input--disabled"
        disabled=""
        id="field_18"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--large field__label--disabled"
        for="field_18"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with small 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--small"
    >
      <input
        class="field__input field__input--inline field__input--small"
        id="field_19"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--small"
        for="field_19"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--small"
    >
      <input
        class="field__input field__input--inline field__input--small"
        id="field_20"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--small"
        for="field_20"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--small field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--small field__input--disabled"
        disabled=""
        id="field_21"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--small field__label--disabled"
        for="field_21"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with success 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--success"
    >
      <input
        class="field__input field__input--inline field__input--success"
        id="field_10"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--success"
        for="field_10"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--success"
    >
      <input
        class="field__input field__input--inline field__input--success"
        id="field_11"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--success"
        for="field_11"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--success field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--success field__input--disabled"
        disabled=""
        id="field_12"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--success field__label--disabled"
        for="field_12"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Checkbox /> renders with touched 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--touched field--success"
    >
      <input
        class="field__input field__input--inline field__input--touched field__input--success"
        id="field_13"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--touched field__label--success"
        for="field_13"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--touched field--success"
    >
      <input
        class="field__input field__input--inline field__input--touched field__input--success"
        id="field_14"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--touched field__label--success"
        for="field_14"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--touched field--success field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--touched field__input--success field__input--disabled"
        disabled=""
        id="field_15"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--touched field__label--success field__label--disabled"
        for="field_15"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_1"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline"
        for="field_1"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_2"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline"
        for="field_2"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--disabled"
        disabled=""
        id="field_3"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--disabled"
        for="field_3"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders with an error 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--erroneous"
    >
      <input
        aria-labelledby="field_4_label field_4_error"
        class="field__input field__input--inline field__input--erroneous"
        id="field_4"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--erroneous"
        for="field_4"
        id="field_4_label"
      >
        Foo
      </label>
      <span
        class="message message--problem"
        id="field_4_error"
        role="alert"
      >
        Oh no!
      </span>
    </div>
    <div
      class="field field--inline field--erroneous"
    >
      <input
        aria-labelledby="field_5_label field_5_error"
        class="field__input field__input--inline field__input--erroneous"
        id="field_5"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--erroneous"
        for="field_5"
        id="field_5_label"
      >
        Bar
      </label>
      <span
        class="message message--problem"
        id="field_5_error"
        role="alert"
      >
        Oh no!
      </span>
    </div>
    <div
      class="field field--inline field--erroneous field--disabled"
    >
      <input
        aria-labelledby="field_6_label field_6_error"
        class="field__input field__input--inline field__input--erroneous field__input--disabled"
        disabled=""
        id="field_6"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--erroneous field__label--disabled"
        for="field_6"
        id="field_6_label"
      >
        Disabled
      </label>
      <span
        class="message message--problem"
        id="field_6_error"
        role="alert"
      >
        Oh no!
      </span>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders with help 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_7"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline"
        for="field_7"
      >
        Foo
        <span
          class="field__help"
        >
          Good luck
        </span>
      </label>
    </div>
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_8"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline"
        for="field_8"
      >
        Bar
        <span
          class="field__help"
        >
          Good luck
        </span>
      </label>
    </div>
    <div
      class="field field--inline field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--disabled"
        disabled=""
        id="field_9"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--disabled"
        for="field_9"
      >
        Disabled
        <span
          class="field__help"
        >
          Good luck
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders with inline 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_22"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline"
        for="field_22"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline"
    >
      <input
        class="field__input field__input--inline"
        id="field_23"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline"
        for="field_23"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--disabled"
        disabled=""
        id="field_24"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--disabled"
        for="field_24"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders with large 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--large"
    >
      <input
        class="field__input field__input--inline field__input--large"
        id="field_16"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--large"
        for="field_16"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--large"
    >
      <input
        class="field__input field__input--inline field__input--large"
        id="field_17"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--large"
        for="field_17"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--large field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--large field__input--disabled"
        disabled=""
        id="field_18"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--large field__label--disabled"
        for="field_18"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders with small 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--small"
    >
      <input
        class="field__input field__input--inline field__input--small"
        id="field_19"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--small"
        for="field_19"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--small"
    >
      <input
        class="field__input field__input--inline field__input--small"
        id="field_20"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--small"
        for="field_20"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--small field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--small field__input--disabled"
        disabled=""
        id="field_21"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--small field__label--disabled"
        for="field_21"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders with success 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--success"
    >
      <input
        class="field__input field__input--inline field__input--success"
        id="field_10"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--success"
        for="field_10"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--success"
    >
      <input
        class="field__input field__input--inline field__input--success"
        id="field_11"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--success"
        for="field_11"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--success field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--success field__input--disabled"
        disabled=""
        id="field_12"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--success field__label--disabled"
        for="field_12"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<RadioGroup /> renders with touched 1`] = `
<fieldset
  class="fieldset fieldset--inline"
>
  <legend
    class="fieldset__legend"
  >
    Jawn
  </legend>
  <div
    class="fieldset__body"
  >
    <div
      class="field field--inline field--touched field--success"
    >
      <input
        class="field__input field__input--inline field__input--touched field__input--success"
        id="field_13"
        name="jawn"
        type="radio"
        value="foo"
      />
      <label
        class="field__label field__label--inline field__label--touched field__label--success"
        for="field_13"
      >
        Foo
      </label>
    </div>
    <div
      class="field field--inline field--touched field--success"
    >
      <input
        class="field__input field__input--inline field__input--touched field__input--success"
        id="field_14"
        name="jawn"
        type="radio"
        value="bar"
      />
      <label
        class="field__label field__label--inline field__label--touched field__label--success"
        for="field_14"
      >
        Bar
      </label>
    </div>
    <div
      class="field field--inline field--touched field--success field--disabled"
    >
      <input
        class="field__input field__input--inline field__input--touched field__input--success field__input--disabled"
        disabled=""
        id="field_15"
        name="jawn"
        type="radio"
        value="buzz"
      />
      <label
        class="field__label field__label--inline field__label--touched field__label--success field__label--disabled"
        for="field_15"
      >
        Disabled
      </label>
    </div>
  </div>
</fieldset>
`;
